cmake/public/protobuf.cmake

```diff
@@ -84,7 +85,7 @@ elseif(Protobuf_FOUND OR PROTOBUF_FOUND)
 endif()
 
 # After above, we should have the protobuf related target now.
-if ((NOT TARGET protobuf::libprotobuf) AND (NOT TARGET protobuf::libprotobuf-lite))
+if ((NOT TARGET protobuf::libprotobuf) AND (NOT TARGET protobuf::libprotobuf-lite) OR (NOT TARGET protobuf::protoc))
```

Comment:
protobuf::protoc probably should not be here, because cmake/public/protobuf.cmake is also used as a configure time file, and when people build libraries depending on Caffe2, they do not necessarily have to have protobuf::protoc present, as long as they do not build custom protobuf files. maybe we should put the check of protobuf::protoc in a separate if statement with a separate warning message.